### PR TITLE
Fix/curriculum en es6 complete a promise - modified test messages to be more clearer

### LINKED ISSUE #38772: Test Error Message does not indicate wrong string
**Describe your problem and how to reproduce it:**

The test error messages do not indicate anything about resolve and reject’s strings.

```
reject should be executed when the if condition is false.
// tests completed
```

This may throw some people off as the message error just indicates the if statement did not fire.

**Add a Link to the page with the problem:**

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject

Test Code:
```
const makeServerRequest = new Promise((resolve, reject) => {
  // responseFromServer represents a response from a server
  let responseFromServer;

  if(responseFromServer) {
    resolve("We String");
  } else {	
    reject("Data String");
  }
});
```

**Tell us about your browser and operating system:**
* Browser Name: 
Chrome
* Browser Version: 
81.0.4044.124

* Operating System: 
iOS 81.0.4044.124

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

BBCE96E4-F281-4C08-9739-F11FCF7C0038
E1BF97DA-2F1F-4B12-8AD0-4DFBEB712881

### PR DESCRIPTION
**Pull request by paulywill**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38772

Upon first doing this challenge the test messages did not indicate the specified string was not matching. This small modification is to redirect the user to carefully compare their string with those indicated in the instructions.

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject.english.md [hunk 1]
```diff
@@ -32,9 +32,9 @@ Make the promise handle success and failure.  If <code>responseFromServer</code>
 
 ```yml
 tests:
-  - text: <code>resolve</code> should be executed when the <code>if</code> condition is <code>true</code>.
+  - text: <code>resolve</code> should be executed with matching string as message when the <code>if</code> condition is <code>true</code>.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject.english.md:35]:
```suggestion
  - text: <code>resolve</code> should be called with the expected string when the <code>if</code> condition is <code>true</code>.
```
Nitpicky, I know, but I wasn't sure what matching or message meant in this context.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject.english.md [hunk 2]
```diff
@@ -32,9 +32,9 @@ Make the promise handle success and failure.  If <code>responseFromServer</code>
...
     testString: assert(removeJSComments(code).match(/if\s*\(\s*responseFromServer\s*\)\s*{\s*resolve\s*\(\s*('|"|`)We got the data\1\s*\)(\s*|\s*;\s*)}/g));
-  - text: <code>reject</code> should be executed when the <code>if</code> condition is <code>false</code>.
+  - text: <code>reject</code> should be executed with matching string as message when the <code>if</code> condition is <code>false</code>.
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject.english.md:37]:
```suggestion
  - text: <code>reject</code> should be called with the expected string when the <code>if</code> condition is <code>false</code>.
```
Same reasoning as above.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I've suggested some tweaks to the wording.  Let me know what you think.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject.english.md (hunk 1 above)
Comment (paulywill) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/complete-a-promise-with-resolve-and-reject.english.md:35]: I concur!

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Pull request merged**